catalog/rocks/validate.go

```diff
@@ -105,12 +108,26 @@ func ValidateTagID(v interface{}) error {
 	if !ok {
 		panic(ErrInvalidType)
 	}
-	if len(s) == 0 {
+	// http://git-scm.com/docs/git-check-ref-format
+	tag := string(s)
+	if len(tag) == 0 || tag == "@" {
 		return ErrRequiredValue
 	}
-	if !reValidBranchID.MatchString(string(s)) {
+	if strings.HasSuffix(tag, ".") || strings.HasSuffix(tag, ".lock") || strings.HasSuffix(tag, "/") {
```

Comment:
I don't understand what a "tag" is.  If this is the user-visible name for a tag, then it seems like this function imposes arbitrary limitations on the vast majority of users, who will never migrate from mvcc.

Reply:
Took the same limitations as git in this case - we can lose some of them as we are git like.

Reply:
Oh, I see!